### PR TITLE
US 1 - List of children

### PR DESCRIPTION
**Pull request by uwoch**
With my app you should be able to record developmental steps of your children. Since it seemed too complicated to me to "create the children" before saving events for them, I decided to use four "sample children" for which events can be saved in one of the next steps.

### CODE REVIEW

File: resources/kids.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+[
+    {
+      "name": "Pixie Fizz",
+      "dob": "01.02.2012",
+      "tooth": "22.07.2012",
```
Comment (moonwave99) [inline resources/kids.json]: `birthDate` over `dob` ^^

I would also use `firstToothDate`.

Also, use ISO Date format, like `2012-02-01` and `2012-07-22`, then you can format it inside your components.

File: components/KidsList/KidsList.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  return (
+    <>
+      <StyledHeading>Alle Meilensteine deines Kindes in einer App!</StyledHeading>
+      <StyledParagraph>Für welches Kind möchtest du loggen?</StyledParagraph>
+      <StyledListContainer>
+      <StyledList>
+        {data.map((kid) => (
+          <StyledListItem key={kid._id}>
+          <StyledImage src={kid.image} alt={kid.name} width={500} height={500} /> <br /> {kid.name} <br /> {kid.dob}
```
Comment (moonwave99) [inline components/KidsList/KidsList.js]: No `<br>` please ^^ make the parent a `flex` column better.

Re `dob`: you can have somewhere in your project a `formatDate` function that takes the timestamp in format `2012-07-22` for instance, and does:

```js
function formatDate(timestamp) {
  return new Date(timestamp).toLocalDateString();
}
```

Then in your components:

```js
<span>{formatDate(kid.dob)}</span>
```

[See docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) for more formatting options.

File: db/models/Kid.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import mongoose from "mongoose";
+const { Schema } = mongoose;
+
+const kidSchema = new Schema({
+  name: { type: String, required: true },
+  dob: { type: String, required: true },
+  tooth: { type: String, required: true },
```
Comment (moonwave99) [inline db/models/Kid.js]: Those should be of type `Date`

File: components/KidsList/KidsList.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import useSWR from "swr";
+import { StyledHeading, StyledList, StyledListContainer,
+    StyledListItem, StyledImage, StyledParagraph } from "./KidsList.styled";
+
+const fetcher = (url) => fetch(url).then((res) => res.json());
+
+export default function KidsList() {
+  const { data, isLoading } = useSWR("/api/kids", fetcher);
```
Comment (moonwave99) [inline components/KidsList/KidsList.js]: You already have the `fetcher` in the SWRConfig ^^

File: pages/api/kids/[id].js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import dbConnect from "../../../db/connect";
+import Kid from "../../../db/models/Kid";
```
Comment (moonwave99) [inline pages/api/kids/[id].js]: Use the same import as below:

```js
import dbConnect from "@/db/connect";
import Kid from "@/db/models/Kid";
```

File: resources/kids.json (hunk 1 above)
Comment (uwoch) [reply resources/kids.json]: Thanks! I will do that.

File: components/KidsList/KidsList.js (hunk 4 above)
Comment (uwoch) [reply components/KidsList/KidsList.js]: Right.

**Pull request merged**